lms/services/canvas_studio.py

```diff
@@ -219,6 +237,53 @@ def list_collection(self, collection_id: str) -> list[File]:
 
         return files
 
+    @classmethod
+    def media_id_from_url(cls, url: str) -> str | None:
+        """Extract the media ID from a `canvas-studio://media/{media_id}` assignment URL."""
+        parsed = urlparse(url)
+        if parsed.scheme != "canvas-studio" or parsed.netloc != "media":
+            return None
+        return parsed.path[1:]
+
+    def get_canonical_video_url(self, media_id: str) -> str:
+        """Return the URL to associate with annotations on a Canvas Studio video."""
+        # We use the REST resource URL as a stable URL for the video.
+        # Example: "https://hypothesis.instructuremedia.com/api/public/v1/media/4"
+        return self._api_url(f"v1/media/{media_id}")
+
+    def get_video_download_url(self, media_id: str) -> str:
+        """Return temporary download URL for a video."""
+
+        download_url = self._api_url(f"v1/media/{media_id}/download")
+        download_rsp = self._oauth_http_service.get(download_url, allow_redirects=False)
```

Comment:
This request can fail if the user added a video from Vimeo or YouTube to Canvas Studio. There is follow-up work to do to present a more helpful error if that happens, as well as to catch the problem earlier, while configuring the assignment.